external/github.go

```diff
@@ -35,21 +38,36 @@ func NewGitHubMatcher(apiURL, token string) (Matcher, error) {
 	if err != nil {
 		return GitHubMatcher{}, err
 	}
-	return GitHubMatcher{client}, nil
+	return GitHubMatcher{
+		client: client,
+		emailToUserCache: make(map[string]string),
+		userToNameCache: make(map[string]string),
+	}, nil
 }
 
 var searchOpts = &github.SearchOptions{
 	Sort:        "joined",
 	ListOptions: github.ListOptions{PerPage: 1},
 }
 
-// MatchByEmail returns the latest GitHub user with the given email.
-func (m GitHubMatcher) MatchByEmail(ctx context.Context, email string) (user, name string, err error) {
-	finished := make(chan struct{})
-	go func() {
-		defer func() { finished <- struct{}{} }()
+// CommitScan
+func (m GitHubMatcher) ScanCommit(ctx context.Context, repo, email, commit string) error {
```

Comment:
Instead of doing this, `FindPeople` could just save the repo and commit for later use in `ReducePeople`.

I think the `CommitterScan` interface and `ScanCommit` function wouldn't be part of the final solution.